Create overdue cron job

cainsusk opened the pull request:
adds a job to notify users when they have items that are due.

> NOTE: changes **Notifications** in the db to **not have an ItemID**, as I thought it would be better to combine all due items into one notification for each user **(lmk if this is bad)**

cainsusk committed:
Create overdue cron job

adds a job to notify users when they have overdue items.

NOTE: changes Notifications in the db to not have an ItemID.

farahwamer reviewed (changes requested):
Works well, but do you mind keeping the itemID column in the notifications table, because I use it in the frontend notifications component to show the item that is overdue or low stock

cainsusk committed:
Fix database

re-adds the item-id column to the Notifications table, and also ensures
loans are confirmed before sending out overdue notifications.

cainsusk commented:
@farahwamer I fixed it (however I made it so that the notification *can* have a null ItemID as the cron-job combines multiple items into one notification (to avoid spamming emails)

#### Testing
use `docker compose build`/`up` 

1. change the job interval (in `app.py`) from `day=*` to `minute=*` (so that the job will occur sooner)
2. The job should complete with no new notifications
3. Add a new loan which is due today (using `adminer` is easiest)
4. check that the job output now says that a notification has been sent
